web/src/widgetize.ts

```diff
@@ -2,15 +2,42 @@ import $ from "jquery";
 
 import * as blueslip from "./blueslip";
 import * as message_lists from "./message_lists";
-
-export const widgets = new Map();
-export const widget_contents = new Map();
-
-export function clear_for_testing() {
+import type {Message} from "./message_store";
+import type {Event, ExtraData as PollWidgetExtraData} from "./poll_widget";
+
+// TODO: This ZFormExtraData type should be moved to web/src/zform.js when it will be migrated
+type ZFormExtraData = {
+    type: string;
+    heading: string;
+    choices: {type: string; reply: string; long_name: string; short_name: string}[];
+};
+
+type WidgetOptions = {
+    widget_type: string;
+    extra_data: PollWidgetExtraData | ZFormExtraData | null;
+    events: Event[];
+    $row: JQuery;
+    message: Message;
+    post_to_server: (data: {msg_type: string; data: string}) => void;
+};
+
+type WidgetValue = Record<string, unknown> & {
+    activate: (data: {
+        $elem: JQuery;
+        callback: (data: string) => void;
```

Comment:
Is the type a `data` parameter always a `string`? It might me for `todo_widget` but for `poll`  `data` looks something like 
```
data: NewOptionOutboundData | QuestionOutboundData | VoteOutboundData | undefined
```
Will be nice if you can cross check and complete this type.